scripts/releases/make-manylinux

```diff
@@ -33,20 +33,27 @@ export BUILD_LIBS=$HOME/.libs
 export CCACHE_DIR="/ccache"
 # Disable some warnings produced by libev especially and also some Cython generated code.
 # Note that changing the value of these variables invalidates configure caches
-GEVENT_WARNFLAGS="-pipe -Wno-strict-aliasing -Wno-comment -Wno-unused-value -Wno-unused-but-set-variable -Wno-sign-compare -Wno-parentheses -Wno-unused-function -Wno-tautological-compare -Wno-strict-prototypes -Wno-return-type -Wno-misleading-indentation"
-export CFLAGS="$GEVENT_WARNFLAGS"
+GEVENT_WARNFLAGS="-Wno-strict-aliasing -Wno-comment -Wno-unused-value -Wno-unused-but-set-variable -Wno-sign-compare -Wno-parentheses -Wno-unused-function -Wno-tautological-compare -Wno-strict-prototypes -Wno-return-type -Wno-misleading-indentation"
+OPTIMIZATION_FLAGS="-pipe"
 if [ "$DOCKER_IMAGE" == "quay.io/pypa/manylinux2014_aarch64" -a -n "$GITHUB_ACTIONS" ]; then
     # Compiling with -Ofast on the arm emulator takes hours. The default settings have -O3,
     # and adding -Os doesn't help much. So maybe -O1 will.
     echo "Compiling with -O1"
-    export CFLAGS="-O1 $GEVENT_WARNFLAGS"
+    OPTIMIZATION_FLAGS="-pipe -O1"
     SLOW_ARM=1
 else
-    echo "Compiling with -Ofast"
-    # Note: -Ofast includes -ffast-math which affects process-wide floating-point flags (e.g. can affect numpy).
-    #       We opt out of -ffast-math explicitly. Other libraries can still trigger it.
-    export CFLAGS="-Ofast -fno-fast-math $GEVENT_WARNFLAGS"
+    echo "Compiling with -O3"
+    # Caution against using -Ofast:
+    # -Ofast includes -ffast-math which affects process-wide floating-point flags (e.g. can affect numpy).
+    # -fno-fast-math disables *some* features of -ffast-math, but as long as -Ofast is specified,
+    # it still affects floating-point flags:
+    # https://github.com/gcc-mirror/gcc/blob/releases/gcc-11.2.0/gcc/config/i386/gnu-user-common.h#L50
+    #
+    # We *could* turn on other flags that are included in -Ofast individually, but for gevent's usecase, it's not
+    # a priority to enable aggressive optimizations at the expense of possibly violating language standards.
```

Comment:
I'd probably remove this comment. We don't need to mention what _not to do_ at every place where we _do_ something. A regression test would be a better way to codify this.

Reply:
I like the idea of a regression test, and thanks for the actual test code!

Why is it alright to skip the test under that condition? In my environment, the skip condition is met, but the assertion fails.

```python
$ python
Python 3.7.12 (default, Sep  4 2021, 05:47:15) 
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gevent
>>> import sys
>>> sys.float_info.max == 1.7976931348623157e+308
True
>>> assert 5e-324 != 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

Reply:
Err, sorry, flip the skip condition :) `==` -> `!=`

The test is whether a subnormal floating point value is equal to zero. What is subnormal depends on the size of float on the platform -- that's why I'm comparing to `float_info.max`: meant for this to be a test for 64-bit platforms.

Reply:
Actually, starting Python 3.9 there's a more portable way to get it, that should obliterate the need for any `skipif`s:
```python
import math

assert math.nextafter(0, 1) != 0
```

Reply:
Ah, thanks. Tried adding the test in a separate branch, which got messy from trying to fix CI failures coming from newer versions of pylint and greenlet. I'll create a separate PR for those fixes and push the test to this branch.

The test did fail in CI (revealing that I missed updating `CFLAGS` used in CI) https://github.com/ento/gevent/runs/8286207244?check_suite_focus=true

Reply:
A regression test has been pushed to this branch, and https://github.com/gevent/gevent/pull/1906 attempts to fix CI failures.